diff --git a/kernel/common/inc/nv-procfs.h b/kernel/common/inc/nv-procfs.h
index e76325c..ac5aea8 100644
--- a/kernel/common/inc/nv-procfs.h
+++ b/kernel/common/inc/nv-procfs.h
@@ -113,9 +113,11 @@ typedef struct file_operations nv_proc_ops_t;
     })
 
 #if defined(NV_PDE_DATA_PRESENT)
-# define NV_PDE_DATA(inode) PDE_DATA(inode)
+#define NV_PDE_DATA(inode) pde_data(inode)
+#elif defined(NV_PDE_DATA_UPPER_CASE_PRESENT)
+#define NV_PDE_DATA(inode) PDE_DATA(inode)
 #else
-# define NV_PDE_DATA(inode) PDE(inode)->data
+#define NV_PDE_DATA(inode) PDE(inode)->data
 #endif
 
 #if defined(NV_PROC_REMOVE_PRESENT)
diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index abbb636..3a83b2b 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -1794,16 +1794,36 @@ compile_test() {
 
 
         pde_data)
+            #
+            # Determine if the pde_data() function is present.
+            #
+            # The commit c28198889c15 removed the function
+            # 'PDE_DATA()', and replaced it with 'pde_data()'
+            # ("proc: remove PDE_DATA() completely") in v5.17-rc1.
+            #
+            CODE="
+            #include <linux/proc_fs.h>
+            void conftest_pde_data(void) {
+                pde_data();
+            }"
+
+            compile_check_conftest "$CODE" "NV_PDE_DATA_PRESENT" "" "functions"
+        ;;
+
+        PDE_DATA)
             #
             # Determine if the PDE_DATA() function is present.
             #
+            # Added by commit d9dda78bad87
+            # ("procfs: new helper - PDE_DATA(inode)") in v3.10
+            #
             CODE="
             #include <linux/proc_fs.h>
             void conftest_PDE_DATA(void) {
                 PDE_DATA();
             }"
 
-            compile_check_conftest "$CODE" "NV_PDE_DATA_PRESENT" "" "functions"
+            compile_check_conftest "$CODE" "NV_PDE_DATA_UPPER_CASE_PRESENT" "" "functions"
         ;;
 
         get_num_physpages)
diff --git a/kernel/nvidia-modeset/nvidia-modeset.Kbuild b/kernel/nvidia-modeset/nvidia-modeset.Kbuild
index 82d1736..8a7645d 100644
--- a/kernel/nvidia-modeset/nvidia-modeset.Kbuild
+++ b/kernel/nvidia-modeset/nvidia-modeset.Kbuild
@@ -90,6 +90,7 @@ NV_CONFTEST_TYPE_COMPILE_TESTS += timeval
 NV_CONFTEST_TYPE_COMPILE_TESTS += proc_dir_entry
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += proc_create_data
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += pde_data
+NV_CONFTEST_FUNCTION_COMPILE_TESTS += PDE_DATA
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += proc_remove
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += timer_setup
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += do_gettimeofday
diff --git a/kernel/nvidia-uvm/nvidia-uvm.Kbuild b/kernel/nvidia-uvm/nvidia-uvm.Kbuild
index d447e23..2f4e1f3 100644
--- a/kernel/nvidia-uvm/nvidia-uvm.Kbuild
+++ b/kernel/nvidia-uvm/nvidia-uvm.Kbuild
@@ -103,6 +103,7 @@ NV_CONFTEST_FUNCTION_COMPILE_TESTS += vzalloc
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += wait_on_bit_lock_argument_count
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += proc_create_data
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += pde_data
+NV_CONFTEST_FUNCTION_COMPILE_TESTS += PDE_DATA
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += proc_remove
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += bitmap_clear
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += usleep_range
diff --git a/kernel/nvidia/nv-mmap.c b/kernel/nvidia/nv-mmap.c
index 288a1e0..d3c751c 100644
--- a/kernel/nvidia/nv-mmap.c
+++ b/kernel/nvidia/nv-mmap.c
@@ -254,6 +254,7 @@ int nv_encode_caching(
         case NV_MEMORY_CACHED:
             if (NV_ALLOW_CACHING(memory_type))
                 break;
+	    __attribute__((__fallthrough__));
         default:
             nv_printf(NV_DBG_ERRORS,
                 "NVRM: VM: cache type %d not supported for memory type %d!\n",
diff --git a/kernel/nvidia/nvidia.Kbuild b/kernel/nvidia/nvidia.Kbuild
index 185d3b5..502f6f0 100644
--- a/kernel/nvidia/nvidia.Kbuild
+++ b/kernel/nvidia/nvidia.Kbuild
@@ -133,6 +133,7 @@ NV_CONFTEST_FUNCTION_COMPILE_TESTS += get_num_physpages
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += efi_enabled
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += proc_create_data
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += pde_data
+NV_CONFTEST_FUNCTION_COMPILE_TESTS += PDE_DATA
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += proc_remove
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += pm_vt_switch_required
 NV_CONFTEST_FUNCTION_COMPILE_TESTS += xen_ioemu_inject_msi
