diff --git a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
index 85d9798..796065f 100644
--- a/kernel/nvidia-drm/nvidia-drm-drv.c
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c
@@ -489,7 +489,7 @@ static int nv_drm_master_set(struct drm_device *dev,
 static void nv_drm_master_set(struct drm_device *dev,
                               struct drm_file *file_priv, bool from_open)
 {
-     WARN_ON(__nv_drm_master_set(dev, file_priv, from_open) != 0);
+     __nv_drm_master_set(dev, file_priv, from_open);
 }
 #endif
 
