diff --git a/kernel/nvidia/nv-mmap.c b/kernel/nvidia/nv-mmap.c
index 288a1e0..65befc5 100644
--- a/kernel/nvidia/nv-mmap.c
+++ b/kernel/nvidia/nv-mmap.c
@@ -254,6 +254,7 @@ int nv_encode_caching(
         case NV_MEMORY_CACHED:
             if (NV_ALLOW_CACHING(memory_type))
                 break;
+           __attribute__((__fallthrough__));
         default:
             nv_printf(NV_DBG_ERRORS,
                 "NVRM: VM: cache type %d not supported for memory type %d!\n",
