diff --git a/kernel/nvidia/nv-acpi.c b/kernel/nvidia/nv-acpi.c
index dba78cf..4f5aff4 100644
--- a/kernel/nvidia/nv-acpi.c
+++ b/kernel/nvidia/nv-acpi.c
@@ -679,7 +679,9 @@ acpi_status nv_acpi_find_methods(
 
 void NV_API_CALL nv_acpi_methods_uninit(void)
 {
+#if defined(NV_ACPI_BUS_GET_DEVICE_PRESENT)
     struct acpi_device *device = NULL;
+#endif
 
     nvif_handle = NULL;
     wmmx_handle = NULL;
