--- a/Makefile	2017-03-30 19:42:29.000000000 +0200
+++ b/Makefile	2017-11-19 23:25:14.220702982 +0100
@@ -1,11 +1,13 @@
 obj-m += i2c-ch341.o
 
+KERN_VER ?= $(shell uname -r)
+
 all:
-		make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
+		make -C /lib/modules/$(KERN_VER)/build M=$(PWD) modules
 
 clean:
-		make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
+		make -C /lib/modules/$(KERN_VER)/build M=$(PWD) clean
 
 reload:	all
 		sudo rmmod i2c-ch341 ||true
-		sudo insmod ./i2c-ch341.ko 
\ No newline at end of file
+		sudo insmod ./i2c-ch341.ko
