Author: Andreas Beckmann <anbe@debian.org>
Description: linux/ioctl32.h is gone since 2.6.22

--- a/common/inc/nv-linux.h
+++ b/common/inc/nv-linux.h
@@ -161,7 +161,8 @@ static inline uid_t __kuid_val(kuid_t ui
 
 #if defined(NVCPU_X86_64) && !defined(HAVE_COMPAT_IOCTL)
 #include <linux/syscalls.h>         /* sys_ioctl()                      */
-#include <linux/ioctl32.h>          /* register_ioctl32_conversion()    */
+// gone since 2.6.22
+//#include <linux/ioctl32.h>          /* register_ioctl32_conversion()    */
 #endif
 
 #if !defined(NV_FILE_OPERATIONS_HAS_IOCTL) && \
--- a/nvidia-modeset/nvidia-modeset-linux.c
+++ b/nvidia-modeset/nvidia-modeset-linux.c
@@ -51,7 +51,8 @@
 
 #if NV_NEEDS_COMPAT_IOCTL_REGISTRATION
 #include <linux/syscalls.h> /* sys_ioctl() */
-#include <linux/ioctl32.h> /* register_ioctl32_conversion() */
+// gone since 2.6.22
+//#include <linux/ioctl32.h> /* register_ioctl32_conversion() */
 #endif
 
 #define NVKMS_MAJOR_DEVICE_NUMBER 195
