--- a/cron.daily	2025-10-02 22:46:04.000000000 +0200
+++ b/cron.daily	2025-10-02 22:54:57.179583467 +0200
@@ -1,10 +1,7 @@
 #!/bin/sh
-set -e
 if [ -e /etc/etckeeper/daily ] && [ -e /etc/etckeeper/etckeeper.conf ]; then
 	. /etc/etckeeper/etckeeper.conf
 	if [ "$AVOID_DAILY_AUTOCOMMITS" != "1" ]; then
-		if ! systemctl --quiet is-enabled etckeeper.timer >/dev/null 2>&1; then
-			/etc/etckeeper/daily
-		fi
+		/etc/etckeeper/daily
 	fi
 fi
