diff --git a/src/vikaggregatelayer.h b/src/vikaggregatelayer.h
index ffe7132..500fe29 100644
--- a/src/vikaggregatelayer.h
+++ b/src/vikaggregatelayer.h
@@ -45,7 +45,7 @@ GType vik_aggregate_layer_get_type ();
 typedef struct _VikAggregateLayer VikAggregateLayer;
 
 void vik_aggregate_layer_uninit ();
-VikAggregateLayer *vik_aggregate_layer_new ();
+VikAggregateLayer *vik_aggregate_layer_new ( VikViewport *vvp );
 void vik_aggregate_layer_add_layer ( VikAggregateLayer *val, VikLayer *l, gboolean allow_reordering );
 void vik_aggregate_layer_insert_layer ( VikAggregateLayer *val, VikLayer *l, GtkTreeIter *replace_layer );
 void vik_aggregate_layer_move_layer ( VikAggregateLayer *val, GtkTreeIter *child_iter, gboolean up );
diff --git a/src/viklayerspanel.c b/src/viklayerspanel.c
index 2ec210a..e4382fd 100644
--- a/src/viklayerspanel.c
+++ b/src/viklayerspanel.c
@@ -537,7 +537,7 @@ static void vik_layers_panel_init ( VikLayersPanel *vlp )
   vlp->hbox = gtk_hbox_new ( TRUE, 2 );
   vlp->vt = vik_treeview_new ( );
 
-  vlp->toplayer = vik_aggregate_layer_new ();
+  vlp->toplayer = vik_aggregate_layer_new ( NULL );
   vik_layer_rename ( VIK_LAYER(vlp->toplayer), _("Top Layer"));
   g_signal_connect_swapped ( G_OBJECT(vlp->toplayer), "update", G_CALLBACK(layers_panel_emit_update), vlp );
 
diff --git a/test/vik2vik.c b/test/vik2vik.c
index 8e2a0ac..ebc5bcb 100644
--- a/test/vik2vik.c
+++ b/test/vik2vik.c
@@ -39,7 +39,7 @@ int main(int argc, char *argv[])
   // Also get lots of warnings about no actual drawing GCs available
   // but for file processing this seems to be good enough
   VikLoadType_t lt;
-  VikAggregateLayer* agg = vik_aggregate_layer_new ();
+  VikAggregateLayer* agg = vik_aggregate_layer_new ( NULL );
   VikViewport* vp = vik_viewport_new ();
 
   lt = a_file_load_stream ( stdin, NULL, agg, vp, NULL, TRUE, FALSE, NULL, "NotUsedName" );
